### PR TITLE
Move libjpeg-turbo utils/impl to imgcodec directory

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

## Category:

**Refactoring**

## Description:

- Moves jpeg_handle.* jpeg_mem.* jpeg_utils.* to imgcodec directory
- Modifies dali/image/CMakeLists.txt to temporarily include those files from the imgcodec directory
- Long term plan is to remove dali/image completely 

### Affected modules and functionalities:

dali/image build system

### Key points relevant for the review:

NA
### Tests:

- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by jantonguirao** [comment]
!build

File: dali/image/CMakeLists.txt [hunk 1]
```diff
@@ -22,3 +22,8 @@ if (NOT BUILD_LIBTIFF)
         ${DALI_SRC_DIR}/dali/image/tiff_libtiff.cc
     )
 endif()
+
+set(DALI_SRCS ${DALI_SRCS}
```
Comment (jantonguirao) [inline dali/image/CMakeLists.txt:26]: Note, `dali/image` is meant to be removed at some point. this is only temporary.

**Comment by jantonguirao** [comment]
!build

File: dali/imgcodec/decoders/CMakeLists.txt [hunk 2]
```diff
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+add_subdirectory(jpeg)
```
Comment (mzient) [inline dali/imgcodec/decoders/CMakeLists.txt:15]: Hmm... so now it only works because we have `-fvisibility=hidden`, doesn't it?

**Review by mzient — APPROVED** [review]

File: dali/imgcodec/decoders/CMakeLists.txt (hunk 2 above)
Comment (jantonguirao) [reply dali/imgcodec/decoders/CMakeLists.txt:15]: I guess so

**Review by stiepan — APPROVED** [review]

**Pull request merged**